src/qiskit_qasm3_import/state.py

```diff
@@ -52,6 +52,16 @@ def physical_qubit_index(name: Union[str, Symbol]) -> Optional[int]:
     return None
 
 
+def add_dummy_parameter_reference(circuit: QuantumCircuit, parameter: Parameter):
+    """Ensure that a circuit contains at least one reference to a given parameter."""
+    # TODO: this is a hack because Terra doesn't have any public way to add a parameter with
+    # no uses to a circuit, but we need to ensure that things work in later bindings if
+    # they're not all there.  This uses the fact that `parameter - parameter` is a
+    # `ParameterExpression` representation of zero that still tracks that it was once
+    # parametric over `parameter`.
+    circuit.global_phase += parameter - parameter
```

Comment:
I guess this potentially does change the type of global phase from a float to parameter expression in Qiskit which is a bit unfortunate. But it should be normalized away if somebody does anything that uses the circuit for anything and global phase gets cast as a float.

Reply:
Yeah, it does, and what's worse, it makes the global phase compare unequal to 0 (because of the parametrisation).  But it should only actually appear for OQ3 programs like
```qasm
input float a;
```
where `a` isn't used, and I think storing the dummy reference in the global phase is probably less disruptive than adding a dummy gate to hold it.